File: day02/aoc2025_02.scala
object Main02 {
    def main(args: Array[String]): Unit = {
        val ranges = scala.io.Source.fromFile("day02/input02")
        .getLines().toList(0).split(',')
        .map(_.split("-").map(_.toLong).toList)
        ranges.take(5).foreach(println)
        
        val (part1, part2) = 
        ranges
        .flatMap { case List(lo, hi) => lo to hi }
        .foldLeft(0L, 0L) { case ((a, b), n) =>
            (a + (if (n.toString.matches("(.+)\\1")) n else 0),
            b + (if (n.toString.matches("(.+)\\1+")) n else 0))
        }
        println("Part 1: " + part1)
        println("Part 2: " + part2)
    }
}

File: day02/aoc2025_02.ts
import { readFileSync } from "fs";

const ranges = readFileSync("day02/input02", 'utf-8')
    .trim().split(",")
    .map(s => s.split('-').map(Number))
    
let part1 = 0
let part2 = 0
for (const [lo, hi] of ranges) {
    for (let n = lo; n <= hi; n++) {
        if (/^(.+)\1$/.test(n.toString())) part1 += n
        if (/^(.+)\1+$/.test(n.toString())) part2 += n
    }
}
console.log("Part 1:", part1)
console.log("Part 2:", part2)
